services/gateway-api/src/circuit_breaker/openai_breaker.py

```
@@ -0,0 +1,362 @@
+        model: str,
+        temperature: float
+    ) -> str:
+        """Generate cache key from request parameters"""
+        # Create a stable hash of the request
+        key_data = {
+            "messages": messages,
+            "model": model,
+            "temperature": temperature
+        }
+        key_str = json.dumps(key_data, sort_keys=True)
+        return hashlib.md5(key_str.encode()).hexdigest()
```

Ruff (0.14.8)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)